core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -243,6 +243,9 @@ public String toString() {
   private final Map<Integer, SortOrder> sortOrdersById;
   private final List<HistoryEntry> snapshotLog;
   private final List<MetadataLogEntry> previousFiles;
+  private final List<SnapshotReference> refs;
```

Comment:
This is where I am a little confused. Does every Metadata.json contain the full branch and snapshot references for all other branches and snapshots?

Reply:
Yes that's the current plan, the metadata.json always contains all the snapshots and all references.

Reply:
If we do so, how are we gonna support snapshot history ? metadata tables per reference should show and use its own snapshots right ?

Reply:
oh, I think we should discuss this in Spec PR, (https://github.com/apache/iceberg/pull/3425)

Reply:
After thinking more on this, I think current snapshot will have parent id. so we can still form a history snapshots per branch just by keeping the current snapshot id. So, no need spec change. 

But we have a change code for handling metadata tables per reference using above information.